HuliWebShop/src/main/java/huli/example/huliwebshop/services/OrderServiceImpl.java

```diff
@@ -32,21 +33,21 @@ public Long createOrder(OrderDTO orderDTO) {
             throw new IllegalArgumentException("User not found with ID: " + orderDTO.getUserId());
         }
 
-        Set<Product> products = new HashSet<>();
-        for (Long productId : orderDTO.getProductIds()) {
+        Set<OrderProduct> products = new HashSet<>();
+        /*for (Long productId : orderDTO.getProductIds()) {
             Product product = iProductRepository.findById(productId).orElse(null);
             if (product != null) {
                 products.add(product);
             }
-        }
+        }*/
```

Comment:
while this part is commented out now, something simipar should happen here, getting the products by the productids peresented in the dto, so we can calculate prices and other things later. however, reaching to the database in a for loop like this is unadvised (this is similar to the N+1 problem - not exactly but it highly resembles to that), because of all the latency it adds to the code and load for the db. There should be a way to get multiple orders with one query, the id-s are already in the dto.